lib/suggestion-list-element.coffee

```diff
@@ -222,16 +213,105 @@ class SuggestionListElement extends HTMLElement
 
     highlightedHTML += replacement.substring(lastWordIndex)
 
-    # Place the snippets back at the placeholders
-    # e.g. highlightedHTML: '<span class="character-match">a</span>bc(|, |)f'
+    # Remove / replace snippet metadata to make it more readable
+    # e.g. highlightedHTML: '<span class="character-match">a</span>bc(${d}, ${e})f'
     # ->   highlightedHTML: '<span class="character-match">a</span>bc(<span class="snippet-completion">d</span>, <span class="snippet-completion">e</span>)f'
-    if snippetCompletions.length
-      completionIndex = 0
-      highlightedHTML = highlightedHTML.replace @snippetMarkerRegex, (match, snippetText) ->
-        snippetCompletions[completionIndex++]
-
+    highlightedHTML = @enhanceSnippet(highlightedHTML) if _.isString(snippet)
     highlightedHTML
 
+  enhanceSnippet: (text) =>
```

Comment:
Can we name this `wrapSnippetsWithHTML` or something similar? 

Why the fat arrow?